c++/src/capnp/serialize.h

```diff
@@ -231,14 +213,6 @@ void writeMessageToFd(int fd, kj::ArrayPtr<const kj::ArrayPtr<const word>> segme
 // =======================================================================================
 // inline stuff
 
-inline FlatArrayMessageReader::FlatArrayMessageReader(
-    kj::ArrayPtr<const word> array, ReaderOptions options)
-#ifdef KJ_DEBUG
-    : UnalignedFlatArrayMessageReader(checkAlignment(array), options) {}
```

Comment:
Can we keep this alignment check in the constructor of `FlatArrayMessageReader`? Otherwise, client code that's sloppy about alignment will not get any indication of its error unless it also happens to call `ListPointer::getStructElement()` and hits the`KJ_DASSERT` there. (I'm assuming that `KJ_DEBUG` is enabled exactly when `KJ_DASSERT` is enabled.)

I still agree with what I wrote in #313. If unaligned buffers are not okay, then we should make the error message for them as clear and as reliable as possible.